src/components/CryptoRichAmI/index.js

```diff
@@ -142,6 +155,7 @@ const Controls = withStyles(controlsStyles)(({ donation, countryCode, coinCode,
             value={donation}
             id='donation'
             onChange={handleIncomeChange}
+            onBlur={() => handleIncomeBlur(donation)}
```

Comment:
It's not really an issue but a compromise. We can't immediately parse the numbers on every input, because if you wanted to type "1.23", once you've typed "1.", the parsing algorithm will parse that string and cast that to a 1, which basically removes your decimal point. The only way I can think of fixing this is by hooking into the focus out event and parse the complete string after it's been completely entered.

Reply:
Ok, found another way instead: use a timeout function to hold off parsing the string input for about 700 milliseconds.